src/network/protocol.rs

```diff
@@ -697,11 +707,36 @@ impl<T: Config> UdpProtocol<T> {
         self.round_trip_time = millis - body.pong;
     }
 
+    /// Upon recveiving a `ChecksumReport`, add it to the checksum history
+    fn on_checksum_report(&mut self, body: &ChecksumReport) {
+        while self.checksum_history.len() > MAX_CHECKSUM_HISTORY_SIZE {
```

Comment:
This should be moved down into the if statement below, and be `>=`. Right now it may sometimes be bigger than `MAX_CHECKSUM_HISTORY_SIZE`.

Alternatively, if @gschup is ok with adding a dependency, we could just use some existing history buffer implementation instead. i.e. https://docs.rs/heapless/latest/heapless/struct.HistoryBuffer.html